airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -0,0 +1,516 @@
+                            "metricHeaderEntries": [
+                                {
+                                    "name": "ga:users",
+                                    "type": "INTEGER"
+                                }
+                            ]
+                        }
+                    },
+                    "data": {
+                        "isDataGolden": true,
+                        "maximums": [
+                            {
+                                "values": [
+                                    "98"
+                                ]
+                            }
+                        ],
+                        "minimums": [
+                            {
+                                "values": [
+                                    "98"
+                                ]
+                            }
+                        ],
+                        "rowCount": 1,
+                        "rows": [
+                            {
+                                "metrics": [
+                                    {
+                                        "values": [
+                                            "98"
+                                        ]
+                                    }
+                                ]
+                            }
+                        ],
+                        "totals": [
+                            {
+                                "values": [
+                                    "98"
+                                ]
+                            }
+                        ]
+                    }
+                }
+            ]
+        }
+
+        Return record which is a map of metric and dimension names and values, like:
+
+        record = {
+                    "view_id":"1111111"
+                    "ga_date":"20210212",
+                    "ga_users":3,
+                    "ga_newUsers":2,
+                    "ga_sessions":7,
+                    "ga_sessionsPerUser":8.0,
+                    "ga_avgSessionDuration":201.0,
+                    "ga_pageviews":43,
+                    "ga_pageviewsPerSession":12.5,
+                    "ga_avgTimeOnPage":83.14035087719298,
+                    "ga_bounceRate":0.0,
+                    "ga_exitRate":6.523809523809524
+        }
+        """
+        json_response = response.json()
+        reports = json_response.get(self.report_field, [])
+
+        for report in reports:
+            column_header = report.get("columnHeader", {})
+            dimension_headers = column_header.get("dimensions", [])
+            metric_headers = column_header.get("metricHeader", {}).get("metricHeaderEntries", [])
+
+            for row in self.get_data(report):
+                record = {}
+                dimensions = row.get("dimensions", [])
+                metrics = row.get("metrics", [])
+
+                for header, dimension in zip(dimension_headers, dimensions):
```

Comment:
`dimension_headers` and `dimensions` are always guaranteed to be in matching order?

Reply:
Yes.